arangod/Agency/Node.h

```diff
@@ -335,16 +212,10 @@ class Node final {
 
   /// @brief accessor to Node's Array
   /// @return  returns nullopt if not found or type doesn't match
-  std::optional<velocypack::Slice> hasAsArray(std::string const&) const;
-
-  // These two operator() functions could be "protected" once
-  //  unit tests updated.
-  //
-  /// @brief Get node specified by path string
-  Node& getOrCreate(std::string_view path);
+  Array const* hasAsArray(std::string const&) const;
```

Comment:
Let's add a code comment here explaining that all `hasAs...` methods walk down the tree and then insist that a certain type is at the end. If not (or if the path does not exist), they return `nullptr` or a `None` option.